src/Generators/staticgenerators.jl

```diff
@@ -0,0 +1,345 @@
+abstract type StaticGenerator <: GraphGenerator end
+
+"""
+    struct Complete <: StaticGenerator
+
+A struct representing a generator for a [complete graph](https://en.wikipedia.org/wiki/Complete_graph)
+with `n` vertices.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the graph.
+"""
+struct Complete{T<:Integer} <: StaticGenerator
+    n::T
+end
+
+"""
+    struct CompleteBipartite <: StaticGenerator
+
+A struct representing a generator for a complete bipartite graph with
+`n1` + `n2` vertices.
+
+### Required Fields
+- `n1::Integer`: the number of vertices in partition 1.
+- `n2::Integer`: the number of vertices in partition 2.
+"""
+struct CompleteBipartite{T<:Integer} <: StaticGenerator
+    n1::T
+    n2::T
+end
+
+"""
+    struct CompleteMultipartite <: StaticGenerator
+
+A struct representing a generator for a complete bipartite graph with
+`sum(pvec)` vertices.
+
+### Required Fields
+- `pvec::AbstractVector{<:Integer}`: a vector with values representing the numbers of vertices in each partition represented by index.
+"""
+struct CompleteMultipartite{T<:Integer, U<:AbstractVector{T}} <: StaticGenerator
+    pvec::U
+    function CompleteMultipartite(pvec::U) where {T<:Integer, U<:AbstractVector{T}}
+        sum(pvec) > typemax(T) && throw(OverflowError("The number of vertices in this graph will overflow its type."))
+        new{T, U}(pvec)
+    end
+end
+
+"""
+    struct Turan <: StaticGenerator
+
+A struct representing a generator for a [Turán Graph](https://en.wikipedia.org/wiki/Tur%C3%A1n_graph), a complete
+multipartite graph with `n` vertices and `r` partitions.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+- `r::Integer`: The number of partitions in the graph
+"""
+struct Turan{T<:Integer, U<:Integer} <: StaticGenerator
```

Comment:
Isn't it a bit overkill to parametrize this generator? Maybe just using `Int` would be enough.

Reply:
This is part of the beauty. In addition to being consistent, we can do this:
```
julia> SimpleGraph(Generators.Turan(0x5, 2))
{5, 6} undirected simple UInt8 graph
```
That is, the type of `n` determines the type of vertex in the produced graph.